src/javascripts/components/navigation.js

```diff
@@ -0,0 +1,144 @@
+import 'govuk-frontend/govuk/vendor/polyfills/Element/prototype/classList'
+
+import common from 'govuk-frontend/govuk/common'
+var nodeListForEach = common.nodeListForEach
+
+var navActiveClass = 'app-navigation--active'
+var navMenuButtonActiveClass = 'govuk-header__menu-button--open'
+var subNavActiveClass = 'app-navigation__subnav--active'
+// This one has the query dot at the beginning because it's only ever used in querySelector calls
+var subNavJSClass = '.js-app-navigation__subnav'
+
+function Navigation ($module) {
+  this.$module = $module || document
+
+  this.$nav = this.$module.querySelector('.js-app-navigation')
+  this.$navToggler = this.$module.querySelector('.js-app-navigation__toggler')
+  this.$navButtons = this.$module.querySelectorAll('.js-app-navigation__button')
+  this.$navLinks = this.$module.querySelectorAll('.js-app-navigation__link')
+
+  // Create variable that saves the last recorded state window size for use by the saveStates function
+  this.savedWindowWidth = null
+}
+
+// Helper function to check if the screen size is above or below the tablet viewport cutoff
+Navigation.prototype.windowSize = function () {
+  return document.documentElement.clientWidth >= 641 ? 'big' : 'small'
+}
+
+// Checks if the saved window size has changed between now and when it was last recorded (on load and on viewport width changes)
+// Reapplies hidden attributes based on if the viewport has changed from big to small or vice verca
+// Saves the new window size
+Navigation.prototype.setHiddenStates = function () {
+  if (this.windowSize() === 'small' && this.savedWindowWidth !== 'small') {
+    this.$nav.setAttribute('hidden', 'true')
```

Comment:
When setting boolean attributes, [you should either set them to the empty string or the attribute name](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute#:~:text=Boolean%20attributes%20are%20considered%20to%20be%20true%20if%20they%27re%20present%20on%20the%20element%20at%20all.%20You%20should%20set%20value%20to%20the%20empty%20string%20(%22%22)%20or%20the%20attribute%27s%20name%2C%20with%20no%20leading%20or%20trailing%20whitespace.).

```suggestion
    this.$nav.setAttribute('hidden', 'hidden')
```